src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -176,17 +176,27 @@ public FireStoreDirectoryEntry retrieveByPath(
         .orElse(null);
   }
 
-  public List<UUID> validateRefIds(TableServiceClient tableServiceClient, List<UUID> refIdArray) {
+  public List<String> validateRefIds(
+      TableServiceClient tableServiceClient, List<String> refIdArray) {
     logger.info("validateRefIds for {} file ids", refIdArray.size());
-    List<UUID> missingIds = new ArrayList<>();
-    for (UUID s : refIdArray) {
-      ListEntitiesOptions options =
-          new ListEntitiesOptions().setFilter(String.format("fileId eq '%s'", s.toString()));
-      if (!TableServiceClientUtils.tableHasEntries(tableServiceClient, TABLE_NAME, options)) {
-        missingIds.add(s);
-      }
-    }
-    return missingIds;
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    return Lists.partition(refIdArray, MAX_FILTER_CLAUSES).stream()
```

Comment:
It looks like we currently use `ListUtils.partition` not `Lists.partition` for this operation.